### ISSUE TITLE
Uncle Blocks?

### ISSUE THREAD

**Issue by olumidesan** [web3coach/the-blockchain-bar#20]
Hi, Lukáš.

This is not an issue, but more like a feature request:

I was wondering if you had plans to integrate uncle blocks into the working of the application. Currently, I know we prevent any forks (`ch_11`), but I've read Ethereum incentivizes uncle blocks for a variety of reasons (at least in its PoW implementation). Would love to see it in action, and was wondering if you had any plans for it in future chapters.

Thanks!

PS: Excellent work, by the way. Absolutely love your book!

**Comment by web3coach** [comment]
Hi Olumide! 

This is an excellent idea! I didn't plan to do it but it's a very interesting topic indeed! How do you feel about submitting a simplified Improvement Proposal on Uncle Blocks and their implementation? ;)

https://github.com/web3coach/the-blockchain-bar/tree/c15_blockchain_forks_what_why_how/TIPs

PS: Glad you like the book!

**Comment by web3coach** [comment]
https://twitter.com/Web3Coach/status/1478046078897381377 :) Will they be still around in PoS?

**Comment by olumidesan** [comment]
Hi, Lukáš.

Sure, I'd love to! However, I just looked at the TIP in the attached link, and it's a little detailed. Consequently, it may take me a bit to come up with a write-up of similar standard. 

Also, referencing your tweet, I had the same concerns (will uncle blocks exist in ETH2.0, and if not, is there any use implementing them here?). That said, I think it'd be fair to have a working PoW implementation with uncle blocks -- **purely** for historical purposes. This repo, in conjunction with your book, was meant, I imagine, as a technical introduction to blockchain and its internal tech. So, it seems apt that uncle blocks -- a key concept in Ethereum's current consensus architecture (GHOST) be discussed.

For all I know, you could have a part two to your book (after ETH2.0 has stabilized, I'd imagine) where you talk about everything new in blockchain, and also the aftermath of uncle blocks. No use vacillating on speculation now.

That's my two cents, anyway :)

**Comment by web3coach** [comment]
It's would definitely take a lot of time to prepare the TIP and create an implementation but you could learn a lot.

Maybe somehow simplifing the exact one in Ethereum?

It would certainly make sense and I would be willing to write a chapter about it. I also wanted to write about libp2p and other p2p libraries for networking but for now I am prioritizing one long tutorial/roadmap that I am releasing in the next weeks.

And yes, I will be adding new concepts from ETH 2.0 like PoS as new/bonus chapters in the book :) The goal is for the book to keep growing and everyone who purchased it to benefit from it.

**Comment by olumidesan** [comment]
I agree it would take time to create the TIP, and I'm open to it, as there's much to learn like you mentioned. I'll see what I can do in the coming days. 

Looking forward to the upcoming chapters!
